### PR TITLE
Adding metric log if httpworker section is present in host.json

### LINKED ISSUE #6083: Emit metrics for custom handlers
Emit a metric log if host.json has httpWorker section

### PR DESCRIPTION
**Pull request by yojagad**
Fixes #6083

Tested this and saw metric event ending up in kusto
image

### CODE REVIEW

File: src/WebJobs.Script/Diagnostics/MetricEventNames.cs [hunk 1]
```diff
@@ -55,6 +55,9 @@ public static class MetricEventNames
         public const string FunctionInvokeSucceeded = "function.invoke.succeeded";
         public const string FunctionInvokeFailed = "function.invoke.failed";
 
+        // Http worker events
+        public const string HttpWorker = "function.worker.httpworkerpresent";
```
Comment (pragnagopa) [inline src/WebJobs.Script/Diagnostics/MetricEventNames.cs]: metric name starting with `function` are meant to be for function invocations or other function level metrics right? Rename to `hostjsonfileconfigurationsource.httpWorker` ?

**Review by pragnagopa — COMMENTED** [review]
Added minor comment.

**Comment by pragnagopa** [comment]
also, can you include httpWorker in https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs#L50 , without this we are not logging host.json httpWorker section

**Comment by pragnagopa** [comment]
Piling on one more - Please add metrics for process start up and also , `DelayUntilWokerInitialized`  - this will help us identify latency after process starts up

File: src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs [hunk 2]
```diff
@@ -39,29 +42,32 @@ internal WorkerProcess(IScriptEventManager eventManager, IProcessRegistry proces
 
         public Task StartProcessAsync()
         {
-            _process = CreateWorkerProcess();
-            try
+            using (_metricsLogger.LatencyEvent(MetricEventNames.ProcessStart))
```
Comment (pragnagopa) [inline src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs:47]: Will this work in webhost scope as well? This method is called in placeholder mode to spin up worker processes.

Comment (yojagad) [reply src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs:47]: Yes, it does. I did the work to unify metric loggers at both scopes long back and what gets injected is this one https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/Diagnostics/NonDisposableMetricsLogger.cs

**Review by pragnagopa — DISMISSED** [review]

**Review by pragnagopa — APPROVED** [review]

**Pull request merged**